staging/src/k8s.io/apiserver/pkg/server/storage/resource_encoding_config.go

```diff
@@ -39,10 +39,7 @@ type DefaultResourceEncodingConfig struct {
 }
 
 type GroupResourceEncodingConfig struct {
-	DefaultExternalEncoding   schema.GroupVersion
```

Comment:
why remove the default group/version targets?

Reply:
It's easier to understand if you take a look at the commit that removes these two fields: https://github.com/kubernetes/kubernetes/pull/67678/commits/a4d4e1e07c3a2711e8b9d8465dc1b83c28515e9e.

There were two ways to set the `DefaultExternalEncoding`.
1. via `SetVersionEncoding`, which was mainly called by tests. I removed all of them and tests still passed. It's also called by `MergeGroupEncodingConfigs`, which was dead code after I removed `--storage-configs`.
2. via `SetResourceEncoding`, which is used to set special encoding version for less stable resources in a group, like `cronjob` stays at `v1beta1` while the `batch` group prefers `v1`. Setting the group's default version to the special resource's version in this case is wrong, because the group's default version is and should be decided by `legacyscheme.Scheme.PreferredVersionAllGroups()`.

Reply:
I vaguely remember @deads2k's saying something along the lines of: "each API type should define its storage version and there should no way to configure it otherwise."  But I may be misremembering.

The ETCD test appears to be passing so I guess this change is fine.

Reply:
My previous comment was rather about how the code was written than explaining why I made the change. I'll try to do the latter now.

Before this PR, there are three layers of "encoding version":
* the default encoding version is stored in the scheme,
* user can override the encoding version of a group via `--storage-version`, the value ends up in `DefaultResourceEncodingConfig.groups[<group>].DefaultExternalEncoding`,
* for less stable resource in a group, e.g., `cronjob` in the `batch` group, we hardcode the less stable resource to be encoded in a different version than the group, see [code](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/default_storage_factory_builder.go#L59-L61). This overriding value ends up in `DefaultResourceEncodingConfig.groups[<group>].ExternalResourceEncodings[<resource name>]`.

In this PR, we removed the `--storage-versions` flag, so the `DefaultExternalEncoding` is not needed. We only need to store the special encoding for those less stable types. This simplified the code in this file a lot.